src/hooks/useUser/index.ts

```diff
@@ -0,0 +1,12 @@
+import { usersApi } from "../../api";
+import { SignUpPayload } from "../../types";
+
+const useUser = () => {
+  const onSignUp = (formData: SignUpPayload) => {
```

Comment:
Esto debería llamarse signup xq es la acción directa. Cuando vos realizas una accion en paralelo por ejemplo podes usar el on. Otros casos serían el on, in, if y demás.